framework/include/materials/MaterialPropertyInterface.h

```diff
@@ -57,10 +58,10 @@ class MaterialPropertyInterface
    *   (3) restricted to only boundaries
    *   (4) restricted to both blocks and boundaries
    */
-  MaterialPropertyInterface(const InputParameters & parameters);
-  MaterialPropertyInterface(const InputParameters & parameters, const std::set<SubdomainID> & block_ids);
-  MaterialPropertyInterface(const InputParameters & parameters, const std::set<BoundaryID> & boundary_ids);
-  MaterialPropertyInterface(const InputParameters & parameters, const std::set<SubdomainID> & block_ids, const std::set<BoundaryID> & boundary_ids);
+  MaterialPropertyInterface(const MooseObject * this_object);
+  MaterialPropertyInterface(const MooseObject * this_object, const std::set<SubdomainID> & block_ids);
+  MaterialPropertyInterface(const MooseObject * this_object, const std::set<BoundaryID> & boundary_ids);
+  MaterialPropertyInterface(const MooseObject * this_object, const std::set<SubdomainID> & block_ids, const std::set<BoundaryID> & boundary_ids);
```

Comment:
Are you planning on cleaning up this multiple constructor stuff? You can now dynamic cast the object to Boundary/BlockRestrictable classes and make the necessary calls to get the block/boundary id sets.